### PR TITLE
RSoC 2021 Announcement

### PR DESCRIPTION
**Pull request by XVilka**
cc @Heersin @Basstorm @fangxlmr

If you wish to add links to blogs/webpages - please suggest me the changes

### CODE REVIEW

**Comment by fangxlmr** [comment]
It would be nice to have [this webpage](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/) in the announcement.

**Comment by XVilka** [comment]
@fangxlmr done, please take a look.

File: content/posts/rsoc-2021-announcement/index.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+title: "Rizin Summer of Code 2021 Announcement"
+date: "2021-05-13"
+summary: "An annoncement of the Rizin Summer of Code 2021. Two accepted candidates."
+tags: ["rizin", "rsoc"]
+ShowToc: true
+TocOpen: false
+weight: 2
+---
+
+For many years during our radare2 times we participated in Google Summer of Code. Also, in addition
```
Comment (officialcjunior) [inline content/posts/rsoc-2021-announcement/index.md:12]:
```suggestion
For many years during our times at radare2, we participated in Google Summer of Code. Also, in addition
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+to hosting those we often did our own Summer of Code. To continue the tradition, albeit under a
+completely new name and a few differences we introduce the RSoC 2021.
+[Together with the KeenLab of Tencent](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/) we provide an opportunity for students
+to apply for the internship and to work on improving Rizin. The application period
+continued through all of April with the fierce competition between candidates. Finally, we chose
```
Comment (officialcjunior) [inline content/posts/rsoc-2021-announcement/index.md:17]: I don't think we need the `the` here.
```suggestion
continued through all of April with fierce competition between candidates. Finally, we chose
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+---
+author: "xvilka"
+title: "Rizin Summer of Code 2021 Announcement"
+date: "2021-05-13"
+summary: "An annoncement of the Rizin Summer of Code 2021. Two accepted candidates."
```
Comment (Surendrajat) [inline content/posts/rsoc-2021-announcement/index.md:5]:
```suggestion
summary: "An announcement of the Rizin Summer of Code 2021. Two accepted candidates."
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+Also, in addition to hosting those we often did our own Summer of Code. To continue the tradition, albeit under a
+completely new name and a few differences we introduce the RSoC 2021.
+[Together with the KeenLab of Tencent](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/) we provide an opportunity for students
+to apply for the internship and to work on improving Rizin. The application period
+continued through all of April with fierce competition between candidates. Finally, we chose
+two. Let them introduce themselves, and we wish them luck!
+
+## Intermediate language improvements
+
+Hello, I'm [Heersin](https://github.com/Heersin) from China, an undergradute student majoring in information security. At the very first, I was looking for a handful RE tool on the linux platform and then I met radare2. I have been using it to solve some basic CTF tasks and dumps from some malwares. I found there are some imperfect features (the concept of project, ESIL, search... ) and want to contribute to it. After knowing there is a new fork named Rizin, which aimed at refactoring radare2, I started to get involved.
```
Comment (Surendrajat) [inline content/posts/rsoc-2021-announcement/index.md:22]:
```suggestion
Hello, I'm [Heersin](https://github.com/Heersin) from China, an undergraduate student majoring in information security. At the very first, I was looking for a handful RE tool on the Linux platform, and then I met radare2. I have been using it to solve some basic CTF tasks and dumps from some malwares. I found there are some imperfect features (the concept of project, ESIL, search...) and want to contribute to it. After knowing there is a new fork named Rizin, which aimed at refactoring radare2, I started to get involved.
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+   3. Add support for `luac` format
+   4. Extend the testsuite to cover more platforms
+
+For RSoC this year, I will be working on the ESIL and follow the [issue 277](https://github.com/rizinorg/rizin/issues/277) to refactor it, and will add support for floating point and bitvectors. I will also try to fix some issues in ESIL (e.g. prevent rizin from getting stuck when hitting some C-library functions).
+
+It will be an exciting and challenging summer, looking forward to it!
+
+## Types analysis
+
+Hi, I'm [basstorm](https://github.com/Basstorm) from China and I am a 21 years old student. Over the last couple of weeks, I have done some bug fixes and improved the class analysis module:
```
Comment (Surendrajat) [inline content/posts/rsoc-2021-announcement/index.md:36]:
```suggestion
Hi, I'm [basstorm](https://github.com/Basstorm) from China, and I am a 21 years old student. Over the last couple of weeks, I have done some bug fixes and improved the class analysis module:
```

**Review by Surendrajat — CHANGES_REQUESTED** [review]

**Comment by wargio** [comment]
Maybe run codespell ? https://github.com/rizinorg/book/issues/46

**Comment by officialcjunior** [comment]
Tried running locally, just now. No misspells found.

File: content/posts/rsoc-2021-announcement/index.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+weight: 2
+---
+
+For many years during our times at radare2 we participated in Google Summer of Code as an organization.
+Also, in addition to hosting those we often did our own Summer of Code. To continue the tradition, albeit under a
+completely new name and a few differences we introduce the RSoC 2021.
+[Together with the KeenLab of Tencent](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/) we provide an opportunity for students
+to apply for the internship and to work on improving Rizin. The application period
+continued through all of April with fierce competition between candidates. Finally, we chose
+two. Let them introduce themselves, and we wish them luck!
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:18]: I would go for another structure -
Starting with "We are excited to announce RSoC 2021. Rizin summer of code is... Together with the KeenLab of Tencent.... we provide the opportunity...."
Then "We use the experience we gathered by participating in Google Summer of Code as an organization and use it to...."
And then the closure sentences about the process and selected students

File: content/posts/rsoc-2021-announcement/index.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+## Intermediate language improvements
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:20]:
```suggestion
## Heersin [fullname]: Intermediate language improvements
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+   1. Update some out-of-date pages in the rizin documentation and add more examples.
+   2. Fix some bugs in `pyc` plugin
+   3. Add support for `luac` format
+   4. Extend the testsuite to cover more platforms
+
+For RSoC this year, I will be working on the ESIL and follow the [issue 277](https://github.com/rizinorg/rizin/issues/277) to refactor it, and will add support for floating point and bitvectors. I will also try to fix some issues in ESIL (e.g. prevent rizin from getting stuck when hitting some C-library functions).
+
+It will be an exciting and challenging summer, looking forward to it!
+
+## Types analysis
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:34]:
```suggestion
## basstorm [or fullname]: Types analysis
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+Hello, I'm [Heersin](https://github.com/Heersin) from China, an undergraduate student majoring in information security. At the very first, I was looking for a handful RE tool on the Linux platform, and then I met radare2. I have been using it to solve some basic CTF tasks and dumps from some malwares. I found there are some imperfect features (the concept of project, ESIL, search...) and want to contribute to it. After knowing there is a new fork named Rizin, which aimed at refactoring radare2, I started to get involved.
+
+During my spare time, I've done some work for rizin, including:
+   1. Update some out-of-date pages in the rizin documentation and add more examples.
+   2. Fix some bugs in `pyc` plugin
+   3. Add support for `luac` format
+   4. Extend the testsuite to cover more platforms
+
+For RSoC this year, I will be working on the ESIL and follow the [issue 277](https://github.com/rizinorg/rizin/issues/277) to refactor it, and will add support for floating point and bitvectors. I will also try to fix some issues in ESIL (e.g. prevent rizin from getting stuck when hitting some C-library functions).
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:30]:
```suggestion
For RSoC this year, I will be working on the ESIL and follow the [issue #277](https://github.com/rizinorg/rizin/issues/277) to refactor it, and will add support for floating point and bitvectors. I will also try to fix some issues in ESIL (e.g. prevent rizin from getting stuck when hitting some C-library functions).
```

**Review by ITAYC0HEN — CHANGES_REQUESTED** [review]
Good work! Added some comments

File: content/posts/rsoc-2021-announcement/index.md (hunk 6 above)
Comment (XVilka) [reply content/posts/rsoc-2021-announcement/index.md]: Good suggestion! Implemented it.

**Review by Surendrajat — APPROVED** [review]

File: content/posts/rsoc-2021-announcement/index.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+summary: "An announcement of the Rizin Summer of Code 2021. Two accepted candidates."
+tags: ["rizin", "rsoc"]
+ShowToc: true
+TocOpen: false
+weight: 2
+---
+
+We are excited to announce the RSoC 2021! Rizin Summer of code is a summer
+internship program. [Together with the KeenLab of Tencent](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/) we provide an opportunity
+for students to work full-time on the Rizin and RzGhidra code.
```
Comment (officialcjunior) [inline content/posts/rsoc-2021-announcement/index.md:14]: I have a small, final opinion. I think this is a bit cleaner?

```suggestion
We are excited to announce the RSoC 2021! Rizin Summer of code is a summer
internship program. [Together with the KeenLab of Tencent](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/) we provide an opportunity
for students to work full-time on Rizin and RzGhidra.
```

**Review by officialcjunior — APPROVED** [review]

File: content/posts/rsoc-2021-announcement/index.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+title: "Rizin Summer of Code 2021 Announcement"
+date: "2021-05-13"
+summary: "An announcement of the Rizin Summer of Code 2021. Two accepted candidates."
+tags: ["rizin", "rsoc"]
+ShowToc: true
+TocOpen: false
+weight: 2
+---
+
+We are excited to announce the RSoC 2021! Rizin Summer of code is a summer
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:12]:
```suggestion
We are excited to announce RSoC 2021! Rizin Summer of Code is a summer
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+internship program. [Together with the KeenLab of Tencent](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/) we provide an opportunity
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:13]:
```suggestion
internship program we organize together with  [KeenLab of Tencent](https://keenlab.tencent.com/zh/2021/03/04/RSoC-keenlab-2021/). We provide an opportunity
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+for students to work full-time on Rizin and RzGhidra.
+We used experience gathered by participating in Google Summer of Code as an organization and
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:15]:
```suggestion
We use the experience we gathered by participating in Google Summer of Code as an organization and
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+organizing our own RSoC as part of the radare2 project.
+
+The application period continued through all of April with fierce competition between candidates. Finally, we chose
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:18]:
```suggestion
The application period continued through all of April and in its end we finally chose
```

File: content/posts/rsoc-2021-announcement/index.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+two students. Let them introduce themselves, and we wish them luck!
```
Comment (ITAYC0HEN) [inline content/posts/rsoc-2021-announcement/index.md:19]:
```suggestion
two students. We wish them the best of luck and happy to give them this stage to introduce themselves.
```

**Review by ITAYC0HEN — CHANGES_REQUESTED** [review]

**Comment by XVilka** [comment]
@ITAYC0HEN applied all your suggestions.

**Review by ITAYC0HEN — APPROVED** [review]
❤️

**Pull request merged**